File: 프로그래머스/2/42578. 의상/의상.js
function solution(clothes) {
    var answer = 0;
    
    let obj = {};
    
    for(const item of clothes) {
        if(obj[item[1]]) obj[item[1]] +=1;
        else obj[item[1]] = 1;
    }
    
    let total = 1;
    
    for(const key in obj) {
        total *= obj[key] +1;
    }
    
    answer = total -1;
    
    return answer;
}

File: 프로그래머스/2/42578. 의상/의상.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

int solution(vector<vector<string>> clothes) {
    int answer = 0;
    
    map<string, int> m;
    
    for(auto& item : clothes)
    {
        m[item[1]] +=1;
    }
    
    int total = 1;
    
    for(auto it = m.begin(); it != m.end(); it++) 
    {
        total *= it->second +1;
    }
    
    total -=1;
    
    answer = total;
    
    return answer;
}